### PR TITLE
fix(ecam): correct registration of new ecam pages

### PR DESCRIPTION
**Pull request by tracernz**
## Summary of Changes
The master branch currently generates an unhandled exception when the ECAM displays load:
image
This was caused by incorrect calls to `registerInstrument` rather than `customElements.define`. Thanks to @Benjozork for pointing me in the right direction.

## Screenshots (if necessary)

image

## Additional context

Discord username (if different from GitHub):

## Testing instructions

- Test ECAM functionality in general
- Test door page works when calling jetway

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

File: src/instruments/ecam-page-template/template.js [hunk 1]
```diff
@@ -36,36 +36,40 @@
...
-/* global BaseInstrument */
-/* global registerInstrument */
+/* global Airliners */
 
-// eslint-disable-next-line camelcase
-class A32NX_PAGE_NAME_Logic extends Airliners.EICASTemplateElement {
-    get templateID() {
-        return 'A32NX_PAGE_NAME_TEMPLATE';
-    }
+let A32NX_PAGE_NAME;
```
Comment (beheh) [inline src/instruments/ecam-page-template/template.js:41]: Isn't this a dangerous name to use as part of the template.js? As far as I understand the string `PAGE_NAME` will be replaced by `door` through our SD rollup.js, so probably any kind of variable should stay well clear of that string?

File: src/instruments/ecam-page-template/template.js [hunk 2]
```diff
@@ -36,36 +36,40 @@
...
-    update(_deltaTime) {
-        this.dispatchEvent(new CustomEvent('update', { detail: this.deltaTime }));
+        update(_deltaTime) {
+            this.dispatchEvent(new CustomEvent('update', { detail: this.deltaTime }));
+        }
     }
-}
+    // eslint-disable-next-line camelcase
+    A32NX_PAGE_NAME.Page = A32NX_PAGE_NAME_Logic;
+})(A32NX_PAGE_NAME || (A32NX_PAGE_NAME = {}));
```
Comment (beheh) [inline src/instruments/ecam-page-template/template.js:73]: I don't get why this has to be an IIFE.

**Comment by tracernz** [comment]
Generated code for door page:
```js
let A32NX_door_page;
((A32NX_door_page) => {
    // eslint-disable-next-line camelcase
    class A32NX_door_page_Logic extends Airliners.EICASTemplateElement {
        get templateID() {
            return 'A32NX_door_page_TEMPLATE';
        }

        connectedCallback() {
            super.connectedCallback();

            // This is big hack, see `template.html`.
            {
                const code = document.getElementById('A32NX_BUNDLED_STYLE').innerHTML;
                const style = document.createElement('style');
                style.innerHTML = code;
                document.head.appendChild(style);
            }
            {
                const code = document.getElementById('A32NX_BUNDLED_LOGIC').innerHTML;
                const script = document.createElement('script');
                script.innerHTML = code;
                document.body.appendChild(script);
            }
        }

        update(_deltaTime) {
            this.dispatchEvent(new CustomEvent('update', { detail: this.deltaTime }));
        }
    }
    // eslint-disable-next-line camelcase
    A32NX_door_page.Page = A32NX_door_page_Logic;
})(A32NX_door_page || (A32NX_door_page = {}));

customElements.define('a32nx-door-page-element', A32NX_door_page.Page);
```

File: src/instruments/ecam-page-template/template.js (hunk 1 above)
Comment (tracernz) [reply src/instruments/ecam-page-template/template.js:41]: It ends up being `A32NX_door_page` which I think is reasonably safe.

File: src/instruments/ecam-page-template/template.js (hunk 2 above)
Comment (tracernz) [reply src/instruments/ecam-page-template/template.js:73]: Can't say I understand why. Just following every other page and figured my JS knowledge is fairly weak.

Comment (davidwalschots) [reply src/instruments/ecam-page-template/template.js:73]: @Benjozork what is your thought on the matter?

Comment (tracernz) [reply src/instruments/ecam-page-template/template.js]: IIFE gone, seems to work just fine as expected. Thanks for the feedback, and now I'm slightly less of a JS noob. 👍

**Review by davidwalschots — APPROVED** [review]
LGTM

**Comment by HugoTro** [comment]
Category: QA Tester
Name: Hugo #3307
Date of testing: Current date 03/03/2021
Version of the sim: 1.13.17.0
PR Tested: #3753 
Tier of testing: 1
Changes to observe: General ECAM functionnality, DOOR page

Testing technique: Description of how you tested the Pull Request
Steps you've made to test the Pull Request:
1. Spawned in C&D
2. Powered the aircraft up using EXT pwr & BATs
3. Checked all ECAM pages while using their appropriate functions (eng while starting engine, door while calling services, etc)

Issues (If Any): None
Steps to achieve the issue

Overall Rating: Good
Conclusions of the testing: LGTM 👍

**Comment by beheh** [comment]
Very small fix, should improve developer QoL. Merging.

**Pull request merged**